mne/io/boxy/boxy.py

```diff
@@ -0,0 +1,392 @@
+        for i_type in data_types:
+            for i_coord in range(len(source_coords[0:total_chans])):
+                boxy_coords.append(np.mean(
+                    np.vstack((source_coords[i_coord], detect_coords[i_coord])),
+                    axis=0).tolist() + source_coords[i_coord] + 
+                    detect_coords[i_coord] + [chan_wavelength[i_coord]] + [0] + [0])
+                boxy_labels.append('S' + 
+                                       str(unique_source_labels.index(source_label[i_coord])+1)
+                                       + '_D' + 
+                                       str(unique_detect_labels.index(detect_label[i_coord])+1) 
+                                       + ' ' + chan_wavelength[i_coord] + ' ' + i_type)
+        
+        # add extra column for triggers
+        boxy_labels.append('Markers')
+
+        ###montage only wants channel coords, so need to grab those, convert to###
+        ###array, then make a dict with labels###
+        for i_chan in range(len(boxy_coords)):
+            boxy_coords[i_chan] = np.asarray(boxy_coords[i_chan],dtype=np.float64) 
+            
+        for i_chan in range(len(all_coords)):
+            all_coords[i_chan] = np.asarray(all_coords[i_chan],dtype=np.float64) 
+
+        all_chan_dict = dict(zip(all_labels,all_coords))
+
+         ###make our montage###
+        my_dig_montage = make_dig_montage(ch_pos=all_chan_dict,
+                                        coord_frame='unknown',
+                                        nasion = fiducial_coords[0],
+                                        lpa = fiducial_coords[1], 
+                                        rpa = fiducial_coords[2])
+        
+        ###create info structure###
+        info = create_info(boxy_labels, srate, ch_types='fnirs_raw')
+        ###add dig info###
+        ## this also applies a transform to the data into neuromag space based on fiducials
+        info.set_montage(my_dig_montage)
+
+        # Store channel, source, and detector locations
+        # The channel location is stored in the first 3 entries of loc.
+        # The source location is stored in the second 3 entries of loc.
+        # The detector location is stored in the third 3 entries of loc.
+        # Also encode the light frequency in the structure.
+       
+        ###place our coordinates and wavelengths for each channel###
+        # # These are all in actual 3d individual coordinates, so let's transform them to
+        # # the Neuromag head coordinate frame
+        trans = get_ras_to_neuromag_trans(fiducial_coords[0], 
+                                    fiducial_coords[1], 
+                                    fiducial_coords[2])
+        
+        for i_chan in range(len(boxy_labels)-1):
+            temp_chn = apply_trans(trans,boxy_coords[i_chan][0:3])
+            temp_src = apply_trans(trans,boxy_coords[i_chan][3:6])
+            temp_det = apply_trans(trans,boxy_coords[i_chan][6:9])
```

Comment:
Should be equivalent to (but please check)
```suggestion
            temp_ch_src_det = apply_trans(trans, boxy_coords[i_chan][:9].reshape(3, 3)).ravel()
```